app/helpers/application_helper.rb

```diff
@@ -801,11 +801,15 @@ def payment_settings_url(gateway_type, person, url_params)
 
   def external_plan_service_login_url(marketplace_id)
     if APP_CONFIG.external_plan_service_url && APP_CONFIG.external_plan_service_secret
```

Comment:
You can remove this if-branch. Just ask the get_external_service_link from the plan service. The `NoPlans` should then return empty link (or Error result) if external plan service is not in use.